pkg/plugin/plugin.go

```diff
@@ -0,0 +1,203 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/lima-vm/lima/v2/pkg/usrlocalsharelima"
+)
+
+type Plugin struct {
+	Name        string `json:"name"`
+	Path        string `json:"path"`
+	Description string `json:"description,omitempty"`
+}
+
+func DiscoverPlugins() ([]Plugin, error) {
+	var plugins []Plugin
+	seen := make(map[string]bool)
+
+	dirs := getPluginDirectories()
+
+	for _, dir := range dirs {
+		pluginsInDir, err := scanDirectory(dir)
+		if err != nil {
+			logrus.Debugf("Failed to scan directory %s: %v", dir, err)
+			continue
+		}
+
+		for _, plugin := range pluginsInDir {
+			if !seen[plugin.Name] {
+				plugins = append(plugins, plugin)
+				seen[plugin.Name] = true
+			}
+		}
+	}
+
+	return plugins, nil
+}
+
+func getPluginDirectories() []string {
```

Comment:
This function is not adding the os.Executable() directory in the front of the PATH.

And it is adding the libexec directory at the front; it should be at then back, so users can override it by putting something else on the PATH.